internal/pkg/istio-router-check/cmd/root.go

```diff
@@ -0,0 +1,186 @@
+	OutputDir               string
+	DetailedCoverage        bool
+	Verbosity               int
+}
+
+func NewCmdRoot() (*cobra.Command, error) {
+	rootCmd := &RootCommand{}
+	cmd := &cobra.Command{
+		Use: "istio-router-check",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if _, err := exec.LookPath(envoyRouterCheckTool); err != nil {
+				return fmt.Errorf("missing route table check tool: %w", err)
+			}
+			logger := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{
+				Level: slog.Level(rootCmd.Verbosity),
+			}))
+			ctx := logr.NewContextWithSlogLogger(cmd.Context(), logger)
+			cmd.SetContext(ctx)
+			return nil
+		},
+		RunE:         rootCmd.Run,
+		SilenceUsage: true,
+	}
+	cmd.Flags().StringVarP(&rootCmd.ConfigDir, "config-dir", "c", "", "directory containing virtualservices")
+	cmd.Flags().StringVarP(&rootCmd.TestDir, "test-dir", "t", "", "directory containing tests")
+	cmd.Flags().BoolVarP(&rootCmd.Details, "details", "d", true, "print detailed information about the test results")
+	cmd.Flags().BoolVarP(&rootCmd.DisableDeprecationCheck, "disable-deprecation-check", "", true, "disable deprecation check")
+	cmd.Flags().BoolVarP(&rootCmd.OnlyShowFailures, "only-show-failures", "", false, "only show failures")
+	cmd.Flags().Float64VarP(&rootCmd.FailThreshold, "fail-threshold", "f", 0.0, "threshold for failure")
+	cmd.Flags().BoolVarP(&rootCmd.CoverageAll, "covall", "", false, "measure coverage by checking all route fields")
+	cmd.Flags().StringVarP(&rootCmd.OutputDir, "output-dir", "o", "", "output directory for coverage information")
+	cmd.Flags().BoolVarP(&rootCmd.DetailedCoverage, "detailed-coverage", "", false, "print detailed coverage information")
+	cmd.Flags().IntVarP(&rootCmd.Verbosity, "", "v", 1, "log verbosity level")
+
+	requiredFlags := []string{"config-dir", "test-dir"}
+	for _, flag := range requiredFlags {
+		if err := cmd.MarkFlagRequired(flag); err != nil {
+			return nil, fmt.Errorf("failed to mark flag %q required: %w", flag, err)
+		}
+	}
+
+	return cmd, nil
+}
+
+func (c *RootCommand) Run(cmd *cobra.Command, _ []string) error {
+	log := logr.FromContextOrDiscard(cmd.Context())
+	tempDir, err := os.MkdirTemp("", ".router-check-tool-")
+	if err != nil {
+		return fmt.Errorf("failed to create temp dir: %w", err)
+	}
+
+	testsFile := filepath.Join(tempDir, "tests.json")
+	routeFile := filepath.Join(tempDir, "routes.json")
+
+	if err := c.prepareTests(cmd.Context(), testsFile); err != nil {
+		return fmt.Errorf("failed to prepare tests: %w", err)
+	}
+
+	if err := c.prepareRoutes(cmd.Context(), routeFile); err != nil {
+		return fmt.Errorf("failed to prepare routes: %w", err)
+	}
+
+	args := c.routerCheckFlags(routeFile, testsFile)
+	routerCheck := exec.Command(envoyRouterCheckTool, args...)
+
+	log.V(3).Info("running command", "command", routerCheck.String())
+	out, err := routerCheck.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%s", out)
+	}
```

Comment:
Here we don't emit the error at all - is that intended?

Reply:
cobra itself will show the error to the user, if we print here it would show the error twice.

Reply:
Ah - I normally mute the cobra version and print my own, but if that's intended makes sense :)

Reply:
I mean that the actual value of `err` (that we check as `err != nil`) is not returned.